api/models/fn.go

```diff
@@ -0,0 +1,235 @@
+package models
+
+import (
+	"net/url"
+	"time"
+
+	"github.com/fnproject/fn/api/id"
+	"github.com/go-openapi/strfmt"
+)
+
+var (
+	// these are vars so that they can be configured. these apply
+	// across function & trigger (resource config)
+
+	MaxMemory      uint64 = 8 * 1024 // 8GB
+	MaxTimeout     int32  = 300      // 5m
+	MaxIdleTimeout int32  = 3600     // 1h
+)
+
+// FuncWrapper makes jason purrty for a Func.
+type FnWrapper struct {
+	Fn *Fn `json:"fn"`
+}
+
+// Fn contains information about a function configuration.
+type Fn struct {
+	// ID is the generated resource id.
+	ID string `json:"id" db:"id"`
+	// Name is a user provided name for this fn.
+	Name string `json:"name" db:"name"`
+	// AppID is the ID of the app this fn belongs to.
+	AppID string `json:"app_id" db:"app_id"`
```

Comment:
this field is inconsistent with the swagger which indicates a field for `app_name` and not `app_id`. it would be nice if this were the entire app object but the tradeoff is that requires a join. in any case, since we don't have 'get app by id' endpoint (we want everything name-addressable), this should probably be `app_name` so that users can look it up, possibly with an additional field like `AppID string json:"-" db:"app_id"` to hide the app id from the api, but to store it for use so we can look up the real name underneath.